Runtime/Scripts/BaseMeshSync.cs

```diff
@@ -1775,49 +1786,199 @@ EntityRecord UpdateInstancedEntity(TransformData data)
-            
-                infoRecord.renderer.UpdateAll(data.transforms, infoRecord.go);
+
+            // Find parent for this instance:
+            GameObject instanceRendererParent;
+            if (m_clientObjects.TryGetValue(data.parentPath, out EntityRecord parentRecord))
+            {
+                instanceRendererParent = parentRecord.go;
+            }
+            else
+            {
+                instanceRendererParent = m_rootObject.gameObject;
+
                 Debug.LogWarningFormat(
                     "[MeshSync] No entity found for parent path {0}, using root as parent", data.parentPath);
             }
 
+            switch (InstanceHandling)
+            {
+                case InstanceHandlingType.InstanceRenderer:
+                    UpdateInstanceInfo_InstanceRenderer(data, infoRecord, instanceRendererParent);
+                    break;
+                case InstanceHandlingType.Copies:
+                case InstanceHandlingType.Prefabs:
+                    UpdateInstanceInfo_CopiesAndPrefabs(data, infoRecord, instancedEntityRecord, instanceRendererParent);
+                    break;
+                default:
+                    break;
+            }
+
             return infoRecord;
         }
 
+        private void UpdateInstanceInfo_CopiesAndPrefabs(
+            InstanceInfoData data,
+            InstanceInfoRecord infoRecord,
+            EntityRecord instancedEntityRecord,
+            GameObject instanceRendererParent) {
+            // Make sure the other renderer is removed if this was not a copy or prefab before:
+            if (infoRecord.renderer != null) {
+                DestroyImmediate(infoRecord.renderer);
+                infoRecord.renderer = null;
+            }
+            infoRecord.DeleteInstanceObjects();
+
+            instanceNames.Add(infoRecord.go.name);
+
+            MeshSyncPlayerConfig config = GetConfigV();
+
+            if (!instanceNames.Contains(infoRecord.go.transform.parent.name)) {
+                bool visible = instancedEntityRecord.visibility.visibleInRender;
+
+                GameObject instanceObjectOriginal;
+                if (InstanceHandling == InstanceHandlingType.Copies) {
+                    instanceObjectOriginal = infoRecord.go;
+                }
+                else {
+                    instanceObjectOriginal = GetOrCreatePrefab(data, infoRecord);
+                }
+
+                if (instanceObjectOriginal == null) {
+                    return;
+                }
+
+                foreach (var mat in data.transforms) {
+                    GameObject instancedCopy;
+                    if (InstanceHandling == InstanceHandlingType.Copies) {
+                        instancedCopy = Instantiate(instanceObjectOriginal, instanceRendererParent.transform);
+                    }
+                    else {
+#if UNITY_EDITOR
+                        instancedCopy = (GameObject)PrefabUtility.InstantiatePrefab(instanceObjectOriginal,
```

Comment:
Can't we do the same thing as runtime in the editor ?

Reply:
The PrefabUtility does not exist outside the editor. PrefabUtility.InstantiatePrefab creates objects like GameObject.Instantiate but keeps the prefab link intact which is what we want in the editor but for runtime we don't need that.